debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/BaseChangeRecordEmitter.java

```diff
@@ -5,31 +5,189 @@
+        this.unavailableValuePlaceholderString = new String(connectorConfig.getUnavailableValuePlaceholder());
+        this.oldColumnValues = oldColumnValues;
+        this.newColumnValues = newColumnValues;
         this.table = table;
     }
 
+    @Override
+    protected Object[] getOldColumnValues() {
+        return oldColumnValues;
+    }
+
+    @Override
+    protected Object[] getNewColumnValues() {
+        return newColumnValues;
+    }
+
+    @Override
     protected void emitTruncateRecord(Receiver receiver, TableSchema tableSchema) throws InterruptedException {
         Struct envelope = tableSchema.getEnvelopeSchema().truncate(getOffset().getSourceInfo(), getClock().currentTimeAsInstant());
         receiver.changeRecord(getPartition(), tableSchema, Operation.TRUNCATE, null, envelope, getOffset(), null);
     }
 
+    @Override
+    protected void emitUpdateAsPrimaryKeyChangeRecord(Receiver receiver, TableSchema tableSchema, Struct oldKey,
+                                                      Struct newKey, Struct oldValue, Struct newValue)
+            throws InterruptedException {
+        if (connectorConfig.isLobEnabled()) {
+            final List<Column> reselectColumns = getReselectColumns(newValue);
+            if (!reselectColumns.isEmpty()) {
+                LOGGER.info("Table '{}' primary key changed from '{}' to '{}' via an UPDATE, re-selecting LOB columns {} out of bands.",
+                        table.id(), oldKey, newKey, reselectColumns.stream().map(Column::name).collect(Collectors.toList()));
+
+                final String query = getReselectQuery(reselectColumns, table);
+                final JdbcConfiguration jdbcConfig = connectorConfig.getJdbcConfig();
+                try (OracleConnection connection = new OracleConnection(jdbcConfig, () -> getClass().getClassLoader(), false)) {
+                    if (!Strings.isNullOrBlank(connectorConfig.getPdbName())) {
+                        connection.setSessionToPdb(connectorConfig.getPdbName());
+                    }
+                    connection.prepareQuery(query,
+                            ps -> prepareReselectQueryStatement(ps, table, newKey),
+                            rs -> updateNewValuesFromReselectQueryResults(rs, reselectColumns));
+
+                    // newColumnValues have been updated via re-select, re-create the event's value
+                    //
+                    // NOTE: The conversion of the column data must occur within the current connection's context.
+                    // This is because the converters for LOB may make specific callbacks to the underlying database
+                    // and if the converter is called outside the scope of the current LOB object, the call will
+                    // fail due to database connection being unavailable.
+                    newValue = tableSchema.valueFromColumnData(newColumnValues);
```

Comment:
What will happen if

* the search row has been deleted in between?
* a DDL was executed between the change and this query?

Reply:
@Naros I think we should detect the generally erroneous situation, end in a hard-stop with poper explanation but at the same time make it conditional based on `event.processing.failure.handling.mode`. In that case there will be a clear sign that something is wrong and data might be off but you'd be able to get unblocked.

Reply:
I see, it is ok then.